### PR TITLE
sqrt behaviour for randomsub plus tests

### PR DESCRIPTION
**Pull request by vyzo**
See #287.

### CODE REVIEW

**Review by yusefnapora — APPROVED** [review]
looks good 👍

**Review by dirkmc — APPROVED** [review]

File: randomsub.go [hunk 1]
```diff
@@ -119,6 +123,14 @@ func (rs *RandomSubRouter) Publish(msg *Message) {
...
+		sqrt := int(math.Ceil(math.Sqrt(float64(rs.size))))
+		if sqrt > target {
+			target = sqrt
+		}
+		if target > len(rspeers) {
+			target = len(rspeers)
+		}
 		xpeers := peerMapToList(rspeers)
 		shufflePeers(xpeers)
 		xpeers = xpeers[:RandomSubD]
```
Comment (emberian) [inline randomsub.go:136]: It goes through all the work of calculating target, but it's not actually used - should it replace the use of `RandomSubD` here?

File: randomsub.go [hunk 2]
```diff
@@ -119,6 +123,14 @@ func (rs *RandomSubRouter) Publish(msg *Message) {
 	}
 
 	if len(rspeers) > RandomSubD {
+		target := RandomSubD
+		sqrt := int(math.Ceil(math.Sqrt(float64(rs.size))))
```
Comment (emberian) [inline randomsub.go:127]: This assumes the size of the network is fixed for the lifetime of the router - what's the suggested use of randomsub in the case where the network size is varying?

File: randomsub.go (hunk 1 above)
Comment (vyzo) [reply randomsub.go:136]: ooops, that's a bug! thanks for spotting.

File: randomsub.go (hunk 2 above)
Comment (vyzo) [reply randomsub.go:127]: It is more of a hint actually. Note that we don't recommend using randomsub in real networks, it was done more as an experiment.

File: randomsub.go (hunk 1 above)
Comment (vyzo) [reply randomsub.go:136]: Care for a patch?

**Pull request merged**